### PR TITLE
Cellular: Read '@' char first then send data.

### PR DESCRIPTION
**Pull request by mudassar-ublox**
### Documentation 

----------------------------------------------------------------------------------------------------------------
### Pull request type 

    [] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [] Feature update (New feature / Functionality change / New API)
    [] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results 

    [] No Tests required for this change (E.g docs only update)
    [] Covered by existing mbed-os tests (Greentea or Unittest)
    [] Tests / results supplied as part of this PR
    
    
----------------------------------------------------------------------------------------------------------------
### Reviewers 

----------------------------------------------------------------------------------------------------------------

### CODE REVIEW

File: features/cellular/framework/targets/UBLOX/AT/UBLOX_AT_CellularStack.cpp [hunk 1]
```diff
@@ -201,7 +201,8 @@ nsapi_size_or_error_t UBLOX_AT_CellularStack::socket_sendto_impl(CellularSocket
             }
             _at.cmd_start_stop("+USOWR", "=", "%d%d", socket->id, blk);
 
-            (void)poll(&fhs, 1, 50);
+            rtos::ThisThread::sleep_for(50);
+            _at.read_bytes(&ch, 1);
```
Comment (fahimalavi) [inline features/cellular/framework/targets/UBLOX/AT/UBLOX_AT_CellularStack.cpp]: After the @ prompt reception, wait for a minimum of 50 ms before sending data is not happening.

File: features/cellular/framework/targets/UBLOX/AT/UBLOX_AT_CellularStack.cpp [hunk 2]
```diff
@@ -168,17 +169,16 @@ nsapi_size_or_error_t UBLOX_AT_CellularStack::socket_sendto_impl(CellularSocket
...
+    uint8_t ch = 0;
 
     if (socket->proto == NSAPI_UDP) {
         if (size > UBLOX_MAX_PACKET_SIZE) {
             return NSAPI_ERROR_PARAMETER;
         }
         _at.cmd_start_stop("+USOST", "=", "%d%s%d%d", socket->id, address.get_ip_address(), address.get_port(), size);
 
-        (void)poll(&fhs, 1, 50);
+        rtos::ThisThread::sleep_for(50);
```
Comment (fahimalavi) [inline features/cellular/framework/targets/UBLOX/AT/UBLOX_AT_CellularStack.cpp]: can't we keep (void)poll(&fhs, 1, 50); instead sleep_for ?

File: features/cellular/framework/targets/UBLOX/AT/UBLOX_AT_CellularStack.cpp (hunk 1 above)
Comment (mudassar-ublox) [reply features/cellular/framework/targets/UBLOX/AT/UBLOX_AT_CellularStack.cpp]: added wait before reading @.

File: features/cellular/framework/targets/UBLOX/AT/UBLOX_AT_CellularStack.cpp (hunk 2 above)
Comment (mudassar-ublox) [reply features/cellular/framework/targets/UBLOX/AT/UBLOX_AT_CellularStack.cpp]: polls logic didnt work.

Comment (fahimalavi) [reply features/cellular/framework/targets/UBLOX/AT/UBLOX_AT_CellularStack.cpp]: It is in riotMicro board as well. If it didn't work then it should be replaced.

**Review by fahimalavi — APPROVED** [review]

**Pull request merged**